include/nlohmann/json.hpp

```diff
@@ -2201,7 +2201,15 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
             }
         }
 
-        JSON_THROW(type_error::create(306, "cannot use value() with " + std::string(type_name()), *this));
+        JSON_THROW(type_error::create(306, detail::concat("cannot use value() with ", type_name()), this));
+    }
+
+    template<class ValueType, class RefStringType, typename std::enable_if<
+                 detail::is_getable<basic_json_t, ValueType>::value, int>::type = 0>
+    JSON_HEDLEY_DEPRECATED(3.10.6)
+    ValueType value(const ::nlohmann::json_pointer<RefStringType>& ptr, const ValueType& default_value) const
```

Comment:
What do these 7 new deprecated functions do?  Are these for compatibility with the `json_pointer` being templated on the JSON type?  These look like they're templated on string type, but then I don't understand why they're the ones that are deprecated.

Reply:
Okay, I see they were added in `537c699` which is for backwards compat.  Maybe it's just the template parameter name that's throwing me off here.  Maybe if it were named something like `LegacyJsonType` it would be less confusing to me.  Maybe just a comment would be enough.  Anyway, not a big deal.

Reply:
I guess you're being thrown off by `RefStringType`. That name is more reflective of the final intent than the current state(and it's `RefStringType` to not shadow `StringType`). These deprecated overloads do indeed handle the case where `RefStringType` is a `basic_json`. I should change that to `BasicJsonType`.

Reply:
Exactly.  :)

Reply:
Done.